tests/lib/rules/jsx-key.js

```diff
@@ -39,6 +39,8 @@ ruleTester.run('jsx-key', rule, {
     {code: 'fn()'},
     {code: '[1, 2, 3].map(function () {})'},
     {code: '<App />;'},
+    {code: '[1, 2, 3].map(x => <App {...{ key }} />);'},
+    {code: '[1, 2, 3].map(x => <App {...objectWithKey} />);'},
```

Comment:
this should fail, i think, unless we want to try to track down `objectWithKey` and ensure it is an object literal with a hardcoded `key` property.

Reply:
I'll have to confess my ignorance of ESLint here: I didn't realize that latter option was even possible.

I thought we'd either have to blanket allow or deny such spread objects .... and I'd argue there *is* value in letting a dev say "I use ESLint as a tool to help me catch problems, but I have no problem remembering to put keys in my objects, so I'd prefer ES Lint not bother me with pointless error messages every time I spread an object" :)

But of course, if it's possible to make ESLint do something that would make any dev happy, and only warn about the legitimately problematic cases, that'd be best.  Can you point me to any rules that do something similar so I could get an idea how?

Reply:
When the object is defined in the same file, as a literal, eslint provides the tools to read it as if it was defined inline.

Also, it's often better to wrongly block something than to wrongly allow it.

I'm not sure off the top of my head which rules do similar things; i'd look into our propType detection code.